### ISSUE TITLE
OCaml 5.1.1 support

### ISSUE THREAD

**Issue by kit-ty-kate** [ocaml/merlin#1713]
Context: https://discuss.ocaml.org/t/ocaml-5-1-1-incoming-breaking-change-in-the-marshal-module/13457

As a result of https://github.com/ocaml/ocaml/pull/12734 merlin doesn’t recognize compressed compiler artefacts anymore and fails with:
```
$ ocamlmerlin single locate -position 2:13 -verbosity 0 < main.ml | jq '.value'
"input_value: compressed object, cannot decompress"
```

**Comment by OlivierNicole** [comment]
Maybe related: installing merlin in 5.1.1 and trying open a .ml file in a properly configured Neovim fails with:
```
Neomake: Error during get_list_entries for merlin: Vim(return):Error invoking 'python_execute' on channel 3 (python3-script-host):^@T...m/5.1.1/share/merlin/vim/autoload/merlin.py", line 182, in command2^@    raise Failure(value)^@merlin.Failure: 'abnormal termination'.
```

**Comment by OlivierNicole** [comment]
It’s the one I have

**Comment by OlivierNicole** [comment]
In fact, it happens in the ocaml compiler repository (without proper build artifacts built for Merlin to use), but in a more usual Dune project it works just fine.

**Comment by voodoos** [comment]
> In fact, it happens in the ocaml compiler repository (without proper build artifacts built for Merlin to use), but in a more usual Dune project it works just fine.

You need to run `make clean` and `dune build @libs` to generate Merlin cofniguration for the compiler, but these rules are often broken.

Not sure what your error means however, it looks like it didn't find the plugin at all...

**Comment by OlivierNicole** [comment]
Yeah, my point was that the behaviour when in the compiler repo, without running `dune build @libs`, is strange. Whereas in another project, I have the normal message “No config found for file… try running 'dune build'”. So not a big issue, but a bit suspect? :thinking:

**Comment by voodoos** [comment]
Closing since we released a version of Merlin compatible with 5.1.1.
